duplicate_ghprb_jobs/tests/test_deduper.py

```diff
@@ -0,0 +1,208 @@
+        self.assertEqual(len(builds), 3)
+
+    def test_get_running_builds_1_building(self):
+        data = sample_data(['4'], ['1', '2', '3'])
+        builds = self.deduper.get_running_builds(data)
+        self.assertEqual(len(builds), 1)
+
+    def test_get_running_builds_none_building(self):
+        data = sample_data([], ['1', '2', '3', '4'])
+        builds = self.deduper.get_running_builds(data)
+        self.assertEqual(len(builds), 0)
+
+    def test_description(self):
+        expected = ("[PR #2] Build automatically aborted because "
+                    "there is a newer build for the same PR. See build #1.")
+        returned = self.deduper._aborted_description(1, 2)
+        self.assertEqual(expected, returned)
+
+    def test_get_json_ok(self):
+        data = sample_data(['1'], [])
+        with patch('requests.get', return_value=mock_response(200, data)):
+            response = self.deduper.get_json()
+            self.assertEqual(data, response)
+
+    def test_get_json_bad_response(self):
+        with patch('requests.get', return_value=mock_response(400)):
+            with self.assertRaises(HTTPError):
+                response = self.deduper.get_json()
+
+    def test_stop_build_ok(self):
+        with patch('requests.post', return_value=mock_response(200, '')):
+            response = self.deduper.stop_build('20')
+            self.assertTrue(response)
+
+    def test_stop_build_bad_response(self):
+        with patch('requests.post', return_value=mock_response(400, '')):
+            with self.assertRaises(HTTPError):
+                response = self.deduper.stop_build('20')
+
+    def test_update_desc_ok(self):
+        with patch('requests.post', return_value=mock_response(200, '')):
+            response = self.deduper.update_build_desc('20', 'new description')
+            self.assertTrue(response)
+
+    def test_update_desc_bad_response(self):
+        with patch('requests.post', return_value=mock_response(400, '')):
+            with self.assertRaises(HTTPError):
+                response = self.deduper.update_build_desc(
+                    '20', 'new description')
+
+    @patch(
+        'duplicate_ghprb_jobs.deduper.GhprbOutdatedBuildAborter.stop_build',
+        return_value=True)
+    @patch(
+        ('duplicate_ghprb_jobs.deduper.GhprbOutdatedBuildAborter.'
+         'update_build_desc'),
+        return_value=True)
+    @patch(
+        ('duplicate_ghprb_jobs.deduper.GhprbOutdatedBuildAborter.'
+         '_aborted_description'),
+        return_value='new description')
+    def test_stop_duplicates_with_duplicates(self, mock_desc,
+                                             update_desc,
+                                             stop_build):
+        sample_buid_data = sample_data(['1', '2', '2', '3'], [])
+        build_data = self.deduper.get_running_builds(sample_buid_data)
+        response = self.deduper.stop_duplicates(build_data)
+        stop_build.assert_called_once_with(1)
```

Comment:
should: give a bit more detail (for example, 'out of the 4 builds, there is one duplicate (a "2"), so this call should only be made for that duplicate')